### PR TITLE
Fix #323: Extend activation provider to allow overriding recovery code generation

### LINKED ISSUE #323: Extend activation provider to allow overriding recovery code generation

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/CustomActivationProvider.java [hunk 1]
```diff
@@ -108,15 +108,35 @@ default boolean shouldAutoCommitActivation(Map<String, String> identityAttribute
...
+     *
+     * @param identityAttributes Identity related attributes.
+     * @param customAttributes Custom attributes, not related to identity.
+     * @param activationType Activation type.
+     * @param context Context for passing parameters between activation provider calls.
+     * @return False to prevent generating recovery codes, "null" to let the PowerAuth Server decide, and true to generate recovery codes
+     *         in case that the feature is enabled globally on PowerAuth Server.
+     * @throws PowerAuthActivationException In case of error in custom activation business logic that should terminate the rest of the activation.
+     */
+    default Boolean shouldCreateRecoveryCodes(Map<String, String> identityAttributes, Map<String, Object> customAttributes, ActivationType activationType, Map<String, Object> context) throws PowerAuthActivationException {
```
Comment (banterCZ) [inline powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/CustomActivationProvider.java:124]: Just a philosophical question. I do not like three-state logic and taking care of NPE. Have you consider to return for example enum?

File: pom.xml [hunk 2]
```diff
@@ -83,13 +83,13 @@
         <maven-source-plugin.version>3.2.1</maven-source-plugin.version>
         <maven-war-plugin.version>3.3.2</maven-war-plugin.version>
         <javax.servlet-api.version>4.0.1</javax.servlet-api.version>
-        <spring-boot.version>2.6.6</spring-boot.version>
+        <spring-boot.version>2.6.8</spring-boot.version>
```
Comment (banterCZ) [inline pom.xml:86]: > Carthago delenda est

IMHO such changes are welcomed but should be placed into a separate PR.

**Review by banterCZ — APPROVED** [review]
Approving with little comments to consider.

File: powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/CustomActivationProvider.java (hunk 1 above)
Comment (petrdvorak) [reply powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/CustomActivationProvider.java:124]: I spent ~hour cracking this myself. The point is that in case someone decides to override the method, they typically want to return true / false.
